packages/styled-components/src/sheet/HoistedTag.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import type { HoistedTag, GroupedTag, Tag } from './types';
+import { DefaultGroupedTag } from './GroupedTag';
+import { WindowedTag } from './WindowedTag';
+
+const hoistedRuleRe = /\s*@(font-face|import)/;
```

Comment:
As I've been testing this, I've learned that `@imports` and `@font-face` can't be intermingled - an `@import` after an `@font-face` is ignored, and so far as I've found in [the specs](https://www.w3.org/TR/css3-cascade/#at-import) a sheet has to have 
```
@charset rule (if it exists)
@import rules
anything else
```

So, @kitten if we change to only hoisting `@import` statements, I would think we could do a custom marker for them (related to https://github.com/styled-components/styled-components/pull/2771/files#r330312750 ) that could be an `@import` using some combination of media conditions that are impossible to meet (not sure what yet), with the import string containing the marker? Does that sound reasonable?

Reply:
So @kitten - Basically we only _need_ to hoist `@import` statements, but so far as I can find there isn't a good "marker" that would work to identify what group they're with. So, I'm going to go ahead and take out `font-face` from this regex (and adjust things), try taking out the markers in the hoisted rules, and see what I can figure out for rehydration (in the other thread).